control-plane/csi-driver/src/bin/controller/controller.rs

```diff
@@ -74,6 +78,66 @@ fn parse_protocol(proto: Option<&String>) -> Result<VolumeShareProtocol, Status>
     }
 }
 
+/// Get app node where volume frontend is situated.
+fn get_volume_app_node(volume: &Volume) -> Option<String> {
+    match volume.spec.target.as_ref() {
+        // todo: Might need to deal with multiple app nodes later?
+        Some(target) => target
+            .frontend_nodes
+            .as_ref()
+            .map(|nodes| nodes[0].name.clone()),
```

Comment:
This will crash if vector is empty vec, which can happen
```suggestion
            .and_then(|nodes| nodes.get(0))
            .map(|node| node.name.clone()),
```